library/std/src/os/mod.rs

```diff
@@ -3,78 +3,143 @@
+#[stable(feature = "os", since = "1.0.0")]
+pub use doc::*;
 
 #[cfg(not(doc))]
-#[cfg(target_os = "wasi")]
-pub mod wasi;
-
-#[cfg(target_os = "android")]
-pub mod android;
-#[cfg(target_os = "dragonfly")]
-pub mod dragonfly;
-#[cfg(target_os = "emscripten")]
-pub mod emscripten;
-#[cfg(all(target_vendor = "fortanix", target_env = "sgx"))]
-pub mod fortanix_sgx;
-#[cfg(target_os = "freebsd")]
-pub mod freebsd;
-#[cfg(target_os = "fuchsia")]
-pub mod fuchsia;
-#[cfg(target_os = "haiku")]
-pub mod haiku;
-#[cfg(target_os = "illumos")]
-pub mod illumos;
-#[cfg(target_os = "ios")]
-pub mod ios;
-#[cfg(target_os = "macos")]
-pub mod macos;
-#[cfg(target_os = "netbsd")]
-pub mod netbsd;
-#[cfg(target_os = "openbsd")]
-pub mod openbsd;
-#[cfg(target_os = "redox")]
-pub mod redox;
-#[cfg(target_os = "solaris")]
-pub mod solaris;
-#[cfg(target_os = "vxworks")]
-pub mod vxworks;
+mod imp {
+    // If we're not documenting std then we only expose modules appropriate for the
+    // current platform.
 
-pub mod raw;
+    #[cfg(all(target_vendor = "fortanix", target_env = "sgx"))]
+    #[path = "../fortanix_sgx/mod.rs"]
+    pub mod fortanix_sgx;
+
+    #[cfg(target_os = "hermit")]
+    #[path = "../hermit/mod.rs"]
+    mod unix;
+
+    #[cfg(target_os = "android")]
+    #[path = "../android/mod.rs"]
+    pub mod android;
+    #[cfg(target_os = "dragonfly")]
+    #[path = "../dragonfly/mod.rs"]
+    pub mod dragonfly;
+    #[cfg(target_os = "emscripten")]
+    #[path = "../emscripten/mod.rs"]
+    pub mod emscripten;
+    #[cfg(target_os = "freebsd")]
+    #[path = "../freebsd/mod.rs"]
+    pub mod freebsd;
+    #[cfg(target_os = "fuchsia")]
+    #[path = "../fuchsia/mod.rs"]
+    pub mod fuchsia;
+    #[cfg(target_os = "haiku")]
+    #[path = "../haiku/mod.rs"]
+    pub mod haiku;
+    #[cfg(target_os = "illumos")]
+    #[path = "../illumos/mod.rs"]
+    pub mod illumos;
+    #[cfg(target_os = "ios")]
+    #[path = "../ios/mod.rs"]
+    pub mod ios;
+    #[cfg(target_os = "l4re")]
+    #[path = "../linux/mod.rs"]
+    pub mod linux;
+    #[cfg(target_os = "linux")]
+    #[path = "../linux/mod.rs"]
+    pub mod linux;
+    #[cfg(target_os = "macos")]
+    #[path = "../macos/mod.rs"]
+    pub mod macos;
+    #[cfg(target_os = "netbsd")]
+    #[path = "../netbsd/mod.rs"]
+    pub mod netbsd;
+    #[cfg(target_os = "openbsd")]
+    #[path = "../openbsd/mod.rs"]
+    pub mod openbsd;
+    #[cfg(target_os = "redox")]
+    #[path = "../redox/mod.rs"]
+    pub mod redox;
+    #[cfg(target_os = "solaris")]
+    #[path = "../solaris/mod.rs"]
+    pub mod solaris;
+    #[cfg(unix)]
+    #[path = "../unix/mod.rs"]
+    pub mod unix;
+
+    #[cfg(target_os = "vxworks")]
+    #[path = "../vxworks/mod.rs"]
+    pub mod vxworks;
+
+    #[cfg(target_os = "wasi")]
+    #[path = "../wasi/mod.rs"]
+    pub mod wasi;
+
+    #[cfg(windows)]
+    #[path = "../windows/mod.rs"]
+    pub mod windows;
+}
+#[cfg(not(doc))]
+#[stable(feature = "os", since = "1.0.0")]
+pub use imp::*;
```

Comment:
The stability attributes on these re-exports don't affect the stability of the exported items.